packages/client/src/components/reports/canary/CanaryReportViewer.tsx

```diff
@@ -33,22 +39,54 @@ export default class CanaryReportViewer extends ConnectedComponent<Props, Stores
   }
 
   async componentDidMount(): Promise<void> {
+    let canaryExecutionStatusResponse: CanaryExecutionStatusResponse | undefined = undefined;
     if (this.stores.canaryExecutorStore.canaryExecutionStatusResponse) {
-      this.setState({
-        canaryExecutionStatusResponse: this.stores.canaryExecutorStore.canaryExecutionStatusResponse
-      });
+      canaryExecutionStatusResponse = this.stores.canaryExecutorStore.canaryExecutionStatusResponse;
     } else {
       const executionId = this.props.match.params.executionId;
-      this.setState({
-        canaryExecutionStatusResponse: await kayentaApiService.fetchCanaryExecutionStatusResponse(executionId)
-      });
+      try {
+        canaryExecutionStatusResponse = await kayentaApiService.fetchCanaryExecutionStatusResponse(executionId);
+      } catch (e) {
+        log.error(`Failed to fetch the canaryExecutionStatusResponse for id: ${executionId}`);
+        throw e;
+      }
     }
+
+    let config = canaryExecutionStatusResponse.config;
+    if (!config) {
+      config = await kayentaApiService.fetchCanaryConfig(
+        ofNullable(canaryExecutionStatusResponse.canaryConfigId).orElseThrow(
+          () => new Error('Expected either a canary config id or canary config object to be present')
+        )
+      );
+    }
+
+    this.setState({
+      canaryExecutionStatusResponse,
+      config
+    });
   }
 
   render(): React.ReactNode {
-    return OptionalUtils.mapIfPresentOrElse(
-      Optional.ofNullable(this.state.canaryExecutionStatusResponse),
-      canaryExecutionStatusResponse => <CanaryReport canaryExecutionStatusResponse={canaryExecutionStatusResponse} />,
+    return mapIfPresentOrElse(
+      ofNullable(this.state.canaryExecutionStatusResponse),
+      canaryExecutionStatusResponse => {
+        if (canaryExecutionStatusResponse.complete && canaryExecutionStatusResponse.status !== 'terminal') {
+          // It now safe to assume, to the best of my knowledge, that result, metricSetPairListId will now not be null
+          return (
+            <CanaryRunResult
+              result={canaryExecutionStatusResponse.result as CanaryResult}
+              metricSetPairListId={canaryExecutionStatusResponse.metricSetPairListId as string}
+              config={this.state.config as CanaryConfig}
+              thresholds={(canaryExecutionStatusResponse.canaryExecutionRequest as CanaryExecutionRequest).thresholds}
+            />
+          );
+        } else if (!canaryExecutionStatusResponse.complete) {
+          // TODO canary execution is still running.
+        } else {
+          // TODO terminal canary execution stuff here.
+        }
+      },
       () => <div>SPINNER HERE</div> // TODO
```

Comment:
I think this could be replaced with:
` <div>
            <h5>Loading</h5>
            <ClipLoader />
          </div>`
but it's not essential